packages/snap-controller/src/Search/SearchController.ts

```diff
@@ -337,15 +360,16 @@ export class SearchController extends AbstractController {
 					// TODO: in future, send as an interaction event
 				}
 			},
-			render: (result: Product | Banner) => {
-				if (this.events.product[result.id]?.render) {
+			render: (result?: Product) => {
```

Comment:
I think `result` should not be optional here.

Maybe just need to undo the changes to this method - since we are not using this for no results conditions any longer.